crates/storage/store_db/libmdbx.rs

```diff
@@ -322,6 +322,53 @@ impl StoreEngine for Store {
         txn.commit().map_err(StoreError::LibmdbxError)
     }
 
+    async fn purge_block(&self, block_number: BlockNumber) -> Result<(), StoreError> {
+        let txn = self
+            .db
+            .begin_readwrite()
+            .map_err(StoreError::LibmdbxError)?;
+        let Some(block_hash) = txn
+            .get::<CanonicalBlockHashes>(block_number)
+            .map_err(StoreError::LibmdbxError)?
+        else {
+            // Block must have been already purged
+            txn.commit().map_err(StoreError::LibmdbxError)?;
```

Comment:
Do we need to commit here? Can't we just abort or discard the tx?

Reply:
I had some problems because of this when testing it